Source/Extensions/Blazorise.FluentValidation/FluentValidationHandler.cs

```diff
@@ -0,0 +1,117 @@
+        {
+            try
+            {
+                var validator = CreateValidator( validation );
+
+                validation.NotifyValidationStarted();
+                var context = CreateContext( validation );
+                var result = validator.Validate( context );
+
+                NotifyValidationChanged( result, validation );
+            }
+            catch ( Exception exc )
+            {
+                HandleException( exc, validation );
+            }
+        }
+
+        public async Task ValidateAsync( IValidation validation, object newValidationValue, CancellationToken cancellationToken = default )
+        {
+            try
+            {
+                var validator = CreateValidator( validation );
+
+                validation.NotifyValidationStarted();
+                var context = CreateContext( validation );
+                var result = await validator.ValidateAsync( context );
+
+                NotifyValidationChanged( result, validation );
+            }
+            catch ( Exception exc )
+            {
+                HandleException( exc, validation );
+            }
+        }
+
+        private void HandleException( Exception exception, IValidation validation )
+        {
+            var msg = $"An unhandled exception occurred when validating field name: '{validation.FieldIdentifier.FieldName}'";
+
+            if ( validation.EditContext.Model != validation.FieldIdentifier.Model )
+            {
+                msg += $" of a child object of type: {validation.FieldIdentifier.Model.GetType()}";
+            }
+
+            msg += $" of <EditForm> model type: '{validation.EditContext.Model.GetType()}'";
+
+            throw new InvalidOperationException( msg, exception );
+        }
+
+        private IValidator CreateValidator( IValidation validation )
+        {
+            var model = validation.FieldIdentifier.Model;
+
+            return TryGetValidatorForObjectType( model.GetType() );
+        }
+
+        private ValidationContext<object> CreateContext( IValidation validation )
+        {
+            var selector = new MemberNameValidatorSelector( new[] { validation.FieldIdentifier.FieldName } );
+
+            return new ValidationContext<object>( validation.FieldIdentifier.Model, new PropertyChain(), selector );
+        }
+
+        private void NotifyValidationChanged( ValidationResult result, IValidation validation )
+        {
+            if ( result.IsValid )
+                validation.NotifyValidationStatusChanged( ValidationStatus.Success );
+            else
+            {
+                var messages = result.Errors.Select( e => e.ErrorMessage );
+                validation.NotifyValidationStatusChanged( ValidationStatus.Error, messages );
+            }
+        }
+
+        private IValidator TryGetValidatorForObjectType( Type modelType )
+        {
+            var validatorType = typeof( IValidator<> );
```

Comment:
@David-Moreira, I think for the first version, we can leave it as it is and later we can optimize this by caching types and creating objects.

Reply:
My guess is that you already based yourself on some quite tested version of this, right? Because I remember there was some caveat when instantiating the Fluent Validators, like it wouldn't just work for every single case with just getting the generic type for some reason. I have to take a look at the implementation I have and see if there's something of note, I don't quite remember now. Something to do with inheritance or multiple generic types?

Reply:
Also about the caching, I'm not even sure this needs it? Are these operations even "heavy"?

Reply:
> My guess is that you already based yourself on some quite tested version of this, right? Because I remember there was some caveat when instantiating the Fluent Validators, like it wouldn't just work for every single case with just getting the generic type for some reason. I have to take a look at the implementation I have and see if there's something of note, I don't quite remember now. Something to do with inheritance or multiple generic types?

I really don't remember it.

---

> Also about the caching, I'm not even sure this needs it? Are these operations even "heavy"?

Well, it can happen with each keystroke on the input. So with blazor-server, it can be "heavy"

Reply:
I think it was something like this.
I don't know if it's a dumb use case haha, but anyway it seems like it works.
![image](https://user-images.githubusercontent.com/22283161/187559232-61f62c97-3fd7-4d52-891d-6dc9631f8f80.png)

Altough you have to manually go register it.
![image](https://user-images.githubusercontent.com/22283161/187559216-7b019957-d8fe-4a02-ae05-019666fa5174.png)